File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool compare(const vector<int>& a, const vector<int>& b) {
    return a[1] < b[1];
}

int solution(vector<vector<int>> routes) {
    int answer = 0;
    sort(routes.begin(), routes.end(), compare);
    //cout << routes.size() << endl;
    for(int i = 0; i < routes.size(); i++){
      int start = routes[i][1];
      answer++;
      for(int j = i; j < routes.size()-1; j++){
        if(routes[i+1][0] <= start){
            i++;
            continue;
        }else{
          break;
        }
      }
    }
    return answer;
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        int answer = 0;
        Arrays.sort(routes, (o1, o2) -> o1[0] - o2[0]);
        
        for(int i = 0; i < routes.length; i++){
            int limit = routes[i][1];
            answer++;
            while (i + 1 < routes.length && routes[i + 1][0] <= limit) {
                i++;
                limit = Math.min(limit, routes[i][1]);
            }
        }
        return answer;
    }
}